artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PageReadWriter.java

```diff
@@ -0,0 +1,281 @@
+import java.util.function.Consumer;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.apache.activemq.artemis.api.core.ActiveMQBuffer;
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.buffers.impl.ChannelBufferWrapper;
+import org.apache.activemq.artemis.core.io.SequentialFile;
+import org.apache.activemq.artemis.core.io.SequentialFileFactory;
+import org.apache.activemq.artemis.core.paging.PagedMessage;
+import org.apache.activemq.artemis.core.persistence.StorageManager;
+import org.apache.activemq.artemis.core.server.LargeServerMessage;
+import org.apache.activemq.artemis.utils.DataConstants;
+import org.apache.activemq.artemis.utils.Env;
+import org.jboss.logging.Logger;
+
+public class PageReadWriter {
+
+
+   private static Logger logger = Logger.getLogger(PageReadWriter.class);
+
+   public static final int SIZE_RECORD = DataConstants.SIZE_BYTE + DataConstants.SIZE_INT + DataConstants.SIZE_BYTE;
+
+   private static final byte START_BYTE = (byte) '{';
+
+   private static final byte END_BYTE = (byte) '}';
+
+   //sizeOf(START_BYTE) + sizeOf(MESSAGE LENGTH) + sizeOf(END_BYTE)
+   private static final int HEADER_AND_TRAILER_SIZE = DataConstants.SIZE_INT + 2;
+   private static final int MINIMUM_MSG_PERSISTENT_SIZE = HEADER_AND_TRAILER_SIZE;
+   private static final int HEADER_SIZE = HEADER_AND_TRAILER_SIZE - 1;
+   private static final int MIN_CHUNK_SIZE = Env.osPageSize();
+
+   public interface SuspectFileCallback {
+      void onSuspect(String fileName, int position, int msgNumber);
+   }
+
+   public interface PageRecordFilter {
+      boolean skip(ActiveMQBuffer buffer);
+   }
+
+   public interface ReadCallback {
+      void readComple(int size);
+   }
+
+   public static final PageRecordFilter ONLY_LARGE = (buffer) -> !PagedMessageImpl.isLargeMessage(buffer);
+
+   public static final PageRecordFilter NO_SKIP = (buffer) -> false;
+
+   public static final PageRecordFilter SKIP_ALL = (buffer) -> true;
+
+   public static int writeMessage(PagedMessage message, SequentialFileFactory fileFactory, SequentialFile file) throws Exception {
+      final int messageEncodedSize = message.getEncodeSize();
+      final int bufferSize = messageEncodedSize + SIZE_RECORD;
+      final ByteBuffer buffer = fileFactory.newBuffer(bufferSize);
+      ChannelBufferWrapper activeMQBuffer = new ChannelBufferWrapper(Unpooled.wrappedBuffer(buffer));
+      activeMQBuffer.clear();
+      activeMQBuffer.writeByte(START_BYTE);
+      activeMQBuffer.writeInt(messageEncodedSize);
+      message.encode(activeMQBuffer);
+      activeMQBuffer.writeByte(END_BYTE);
+      assert (activeMQBuffer.readableBytes() == bufferSize) : "messageEncodedSize is different from expected";
+      //buffer limit and position are the same
+      assert (buffer.remaining() == bufferSize) : "buffer position or limit are changed";
+      file.writeDirect(buffer, false);
+      return bufferSize;
+   }
+
+
+
```

Comment:
Its weird that there are all these newlines between methods, but a little below this, there is a 100 line method that is screaming for some newline grouping/seperation but has none.

Reply:
I am just moving what was on Page here. not new code!

It's not my original code, but I will see where I would do some spaces.